server/test/services/applications/ProgramAdminApplicationServiceTest.java

```diff
@@ -66,4 +68,36 @@ public void getApplication_emptyAdminName() {
 
     assertThat(service.getApplication(application.id, program)).isEmpty();
   }
+
+  @Test
+  public void getNote_noNotes_empty() {
+    ProgramDefinition program = ProgramBuilder.newActiveProgram("some-program").buildDefinition();
+
+    Applicant applicant = resourceCreator.insertApplicantWithAccount();
+    Application application =
+        Application.create(applicant, program.toProgram(), LifecycleStage.ACTIVE)
+            .setSubmitTimeToNow();
+
+    // Execute, verify.
+    assertThat(service.getNote(application)).isEmpty();
+  }
+
+  @Test
+  public void getNote_multipleNotes_findsLatest() {
+    String note = "Application note";
+    ProgramDefinition program = ProgramBuilder.newActiveProgram("some-program").buildDefinition();
+
+    Account account = resourceCreator.insertAccount();
+    Applicant applicant = resourceCreator.insertApplicantWithAccount();
+    Application application =
+        Application.create(applicant, program.toProgram(), LifecycleStage.ACTIVE)
+            .setSubmitTimeToNow();
+
+    service.setNote(application, NoteEvent.create("first note"), account);
+    service.setNote(application, NoteEvent.create(note), account);
```

Comment:
I *think* you might end up with a potentially flaky test here since EBean serializes timestamps truncated to millisecond precision. When I was developing similar tests, they passed on my machine, then failed in CI (presumably since the tests ran faster).

If you want to account for this in the test and avoid that type of flake, you could add a sleep for 2ms. I don't really think that would have a dramatic impact on our test runtime and would hopefully reduce the flake rate.

Context: https://github.com/civiform/civiform/blob/08868c0bc54327e3ed4289c9827ba5c471f109e8/server/test/models/QuestionTest.java#L210-L215

Reply:
done!